src/main/java/frc/robot/commands/ArmJoystickConeCommand.java

```diff
@@ -0,0 +1,214 @@
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.button.CommandXboxController;
+import frc.robot.config.ArmConfig;
+import frc.robot.config.ArmConfig.ArmSetpoint;
+import frc.robot.subsystems.ArmDisplay;
+import frc.robot.subsystems.ArmSubsystem;
+import frc.robot.subsystems.ArmWaypoint;
+
+public class ArmJoystickConeCommand extends CommandBase {
+  ArmSetpoint armSetpoint;
+  double defaultAngle1 = Math.PI; // angle in radians of joint 1 in default position (vertical)
+  double defaultAngle2 = Math.PI/36; // angle in radians of joint 2 in default position (angle of 5 degrees)
+  double[] defaultAngles;
+  double[] angles;
+  double angle1;
+  double angle2;
+  int index;
+  double tempX;
+  double tempZ;
+
+  // arm simulation variables
+  ArmDisplay armDisplay;
+
+  boolean startBrakeTimer;
+  Timer m_timer = new Timer();
+  Timer m_timer2 = new Timer();
+
+  // joystick value controlling cone arm
+  private CommandXboxController m_joystick;
+  double z_offset = 0;
+  Debouncer m_topDebounce = new Debouncer(0.3);
+  Debouncer m_bottomDebounce = new Debouncer(0.3);
+  
+
+  /** Creates a new ArmExtend. */
+  
+  public ArmJoystickConeCommand(ArmSetpoint armSetpoint, CommandXboxController operator_stick) {
+    this.armSetpoint = armSetpoint;
+    this.m_joystick = operator_stick;
+    // Use addRequirements() here to declare subsystem dependencies.
+    addRequirements(ArmSubsystem.getInstance());
+  }
+
+
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    System.out.println("~~~~~" + armSetpoint.name());
+
+    if (Double.isNaN(armSetpoint.getX()) || Double.isNaN(armSetpoint.getZ())) {
+      DriverStation.reportError("ArmSetpoint called " + armSetpoint.name() + " is NaN", true);
+      this.cancel();
+    }
+
+    if (armSetpoint.getWaypoint().length == 0) {
+      index = 99;
+    } else {
+      index = 0;
+    }
+    
+    ArmSubsystem.getInstance().resetMotionProfile();
+    ArmSubsystem.getInstance().controlTopArmBrake(false);
+    ArmSubsystem.getInstance().controlBottomArmBrake(false);
+
+    if (armSetpoint == ArmSetpoint.MIDDLE_CONE) {
+      ArmSubsystem.getInstance().setTopConstraints(ArmConfig.TOP_CONE_MIDDLE_MAX_VEL, ArmConfig.TOP_CONE_MIDDLE_MAX_ACCEL);
+    }
+    if (armSetpoint == ArmSetpoint.TOP_CONE) {
+      ArmSubsystem.getInstance().setTopConstraints(ArmConfig.TOP_CONE_TOP_MAX_VEL, ArmConfig.TOP_CONE_TOP_MAX_ACCEL);
+    }
+
+    startBrakeTimer = false;
+    m_timer.stop();
+    m_timer.reset();
+    m_timer2.stop();
+    m_timer2.reset();
+
+    z_offset = 0;
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    ArmWaypoint waypoint;
+
+    if (index >= 99) {
+      tempX = armSetpoint.getX() + z_offset * -0.5;
```

Comment:
Do we need to add ( ) to -0.5?
The same comment to the following z*(-0.16)?

Reply:
I am fairly confidant that Java treats -0.5 as a signed floating point number and is higher in the order of operations.

It worked when we were testing.